src/Filter/FieldDefinitionAdapter/AdvancedManyToManyRelation.php

```diff
@@ -25,4 +25,22 @@ class AdvancedManyToManyRelation extends ManyToOneRelation implements FieldDefin
      */
     protected $fieldType = "advancedManyToManyRelation";
 
+    /**
+     * @inheritDoc
+     */
+    protected function doGetIndexDataValue($object, $ignoreInheritance = false) {
+        $value = parent::doGetIndexDataValue($object, $ignoreInheritance);
+
+        $filteredValues = array_map(function($item) {
+
+            if(isset($item['element'])) {
+                return $item['element'];
+            } else {
+                return $item;
+            }
+        }, $value);
```

Comment:
Not sure what the minimum PHP version of this bundle is... but if it is PHP >= 7.4 it could also be written like this:
```suggestion
        $filteredValues = array_map(static fn($item) => $item['element'] ?? $item, $value);
```